storage/reads/reader.go

```diff
@@ -648,3 +648,27 @@ func (ti *tagValuesIterator) handleRead(f func(flux.Table) error, rs cursors.Str
 func (ti *tagValuesIterator) Statistics() cursors.CursorStats {
 	return cursors.CursorStats{}
 }
+
+type duplicateFilter struct {
+	f    func(tbl flux.Table) error
+	seen *execute.GroupLookup
+}
+
+func filterDuplicateTables(f func(tbl flux.Table) error) func(tbl flux.Table) error {
+	filter := &duplicateFilter{
+		f:    f,
+		seen: execute.NewGroupLookup(),
+	}
+	return filter.Process
+}
+
+func (df *duplicateFilter) Process(tbl flux.Table) error {
+	// Identify duplicate keys within the cursor.
+	if _, ok := df.seen.Lookup(tbl.Key()); ok {
+		// Discard this table.
```

Comment:
Should we produce a log message here?  If this was happening a lot, then it would be a problem.

Reply:
I was wondering that, but the logger didn't seem to have easy access here so I opted not to for now.